Новый админский режим

XxRaay opened the pull request:
## О ПР`е
Добавлен новый режим игры "Бой предателей", где все игроки становятся предателями с целями на выживание.

## Технические детали
<!-- Сводка изменений кода для более удобного просмотра. -->

## Изменения кода официальных разработчиков
<!-- Сводка изменений кода визардов для более удобного просмотра (Если есть). -->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- Новые функции
  - Добавлен режим «Бой предателей»: все игроки — агенты Синдиката с присвоением ролей, целями и подсчётом очков (телекристаллы, убийства, выживание).
  - Стартовая пацифия с настраиваемой длительностью и напоминаниями.
  - Машины искупления для обмена КПК на телекристаллы; спавн машин по станциям.
  - Новая цель — «Сохранить жизнь» для агентов.

- Локализация
  - Полная русская локализация режима, сообщений, итогов раунда и машины искупления.

- Прочее
  - Добавлен интерактивный прототип машины искупления, спрайт и лицензионные сведения.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

XxRaay committed:
feat(gameplay): add SyndieBattle game mode with traitor conversion

Add new SyndieBattle game mode where all players are converted to traitors with objectives. Includes:
- Game rule component and system for managing traitor conversion
- Russian localization for the game mode
- Prototype definitions for objectives and game rules
- Survival objectives and kill objectives for traitors

XxRaay commented:
@coderabbitai rewiew

XxRaay commented:
@coderabbitai review

XxRaay commented:
что за хуйню он высрал

XxRaay commented:
@coderabbitai full review

XxRaay committed:
refactor(SyndieBattle): clean up and enhance game rule system

- Removed unnecessary comments and unused dependencies in SyndieBattleRuleSystem.
- Simplified player conversion logic and improved event handling for traitor selection.
- Added new localization strings for kill scoring and redemption messages.
- Updated prototype to inherit from BaseTraitorRule for better integration with existing systems.

XxRaay committed:
feat(gameplay): add syndicate battle game mode with traitor conversion

- Implement SyndieBattle game mode where all players become traitors
- Add localization for Russian and English
- Modify redemption system to check for uplink component directly
- Convert existing players to traitors when mode starts
- Assign traitor objectives with kill focus

Ser1-1y committed:
fixes to the syndiebattle gamemode

(cherry picked from commit 64abf96d48a1993ab20d0f1135e7456e3f7c47d8)

XxRaay committed:
feat(SyndieBattle): enhance gameplay with redemption machines and kill tracking

- Added SyndieBattle Redemption Machines that allow players to exchange enemy PDAs for TC.
- Implemented kill tracking with detailed kill messages including killer, victim, and location.
- Updated localization for new features and improved existing strings.
- Refactored game rule system to better integrate with the new components.

XxRaay committed:
fix: update subproject commit and modify syndiebattle objective icon

- Updated subproject commit to indicate a dirty state.
- Changed the icon for the syndiebattle objectives from a generic job icon to a specific vehicle sprite.

XxRaay committed:
fix: update subproject commit and change syndiebattle objective icon

- Marked subproject commit as dirty to reflect local changes.
- Updated the syndiebattle objective icon to a specific ghost sprite for better thematic alignment.

XxRaay committed:
feat(SyndieBattle): add redemption machine count and enhance score tracking

XxRaay committed:
fix(SyndieBattleScoreComponent): comment out unused PlayerId field and update subproject commit

XxRaay committed:
Revert "fix(SyndieBattleScoreComponent): comment out unused PlayerId field and update subproject commit"

This reverts commit 82556d88d5f63b5afafe9976248da32a20258a2f.

XxRaay committed:
feat: add license file for content under ICLA

XxRaay commented on Content.Server/Imperial/XxRaay/SyndieBattle/license.txt:
```diff
@@ -0,0 +1,2 @@
+license: This content is under ICLA. Copyright holder: pups2004 (discord) (Read more on: https://wiki.imperialspace.net/ru/icla)
+copyright: "pups2004 (discord)"
```
никогда более не предлагай такую чушь

Ser1-1y commented on Resources/Locale/ru-RU/Imperial/XxRaay/preset-syndiebattle.ftl:
```diff
@@ -0,0 +1,12 @@
+.desc = Машина, которая позволяет обменивать КПК врагов на телекристаллы. Вставьте КПК, чтобы получить телекристаллы.
+
+syndiebattle-kill-detail = { $killer } убил { $victim }. Место: { $location }.
+syndiebattle-redemption-no-pda = Это не КПК!
+syndiebattle-redemption-no-uplink = У этого КПК нет аплинка Синдиката!
+syndiebattle-redemption-success = Успех! Получено предметов: { $amount }
```
```suggestion
syndiebattle-redemption-no-pda = Оно не помещается в машину!
syndiebattle-redemption-no-uplink = Машина вернула КПК!
syndiebattle-redemption-success = Машина выдала { $amount } предметов!
```
Нон рп информация это плохо!!!

Ser1-1y commented on Resources/Locale/ru-RU/Imperial/XxRaay/preset-syndiebattle.ftl:
```diff
@@ -0,0 +1,12 @@
+
+# SyndieBattle Redemption Machine
+ent-SyndieBattleRedemptionMachine = машина искупления
+.desc = Машина, которая позволяет обменивать КПК врагов на телекристаллы. Вставьте КПК, чтобы получить телекристаллы.
```
```suggestion
.desc = Машина с подозрительно похожим на КПК отверстием.
```

Ser1-1y commented on Content.Server/Imperial/XxRaay/SyndieBattle/SyndieBattleRuleSystem.cs:
```diff
@@ -0,0 +1,259 @@
+    /// <summary>
+    /// Получает имя сущности
+    /// </summary>
+    private string GetEntityName(EntityUid entity)
+    {
+    return MetaData(entity).EntityName;
+    }
```
Зачем для этого отдельный метод?

XxRaay replied:
Раньше там был большой код который брал сикей игрока

Ser1-1y replied:
Ну сейчас же нет поэтому убрать отдельный метод лучше

Ser1-1y commented on Content.Server/Imperial/XxRaay/SyndieBattle/SyndieBattleRuleSystem.cs:
```diff
@@ -0,0 +1,259 @@
+        {
+            return "неизвестном месте";
+        }
+    }
```
В чём смысл try catch если и так есть 
``string.IsNullOrEmpty(location) ? "неизвестном месте" : location``

Ser1-1y commented on Content.Server/Imperial/XxRaay/SyndieBattle/SyndieBattleScoreComponent.cs:
```diff
@@ -0,0 +1,17 @@
+    [DataField]
+    public int Score;
+
+    [DataField]
+    public NetUserId? PlayerId;
```
Зачем оно нужно?

Ser1-1y commented on Content.Server/Imperial/XxRaay/SyndieBattle/SyndieBattleRuleSystem.cs:
```diff
@@ -0,0 +1,259 @@
+    /// </summary>
+    private string GetKillerName(KillSource source)
+    {
+        switch (source)
+        {
+            case KillPlayerSource player:
+                if (!_playerManager.TryGetSessionById(player.PlayerId, out var session))
+                    return "Неизвестный игрок";
+                if (session.AttachedEntity == null)
+                    return "Неизвестный игрок";
+                return MetaData(session.AttachedEntity.Value).EntityName;
+
+            case KillNpcSource npc:
+                if (Deleted(npc.NpcEnt))
+                    return "Неизвестный NPC";
+                return MetaData(npc.NpcEnt).EntityName;
+
+            case KillEnvironmentSource:
+                return "Окружение";
+        }
+
+        return "Неизвестно";
```
Мне кажется что раунды этого режима будут короткими, вероятность смерти от других причин очень малы, не бессмысленный ли это код? Я бы оставил смерть от окружения и игрока. "Неизвестно" заменил бы на "Что-то"

Ser1-1y commented on Content.Server/Imperial/XxRaay/SyndieBattle/SyndieBattleRuleSystem.cs:
```diff
@@ -0,0 +1,259 @@
+
+        for (var i = 0; i < ruleComp.RedemptionMachineCount; i++)
+        {
+            if (!TryFindRandomTile(out _, out _, out _, out var coords))
```
Заменить на TryFindRandomTileOnStation иначе спавн происходит на случайной станции, что не идеально.

Нужна проверка на занятость тайла.

Ser1-1y reviewed (commented):
🤓

XxRaay committed:
Update Resources/Locale/ru-RU/Imperial/XxRaay/preset-syndiebattle.ftl

Co-authored-by: Ser11y <160628372+Ser1-1y@users.noreply.github.com>

XxRaay committed:
Update Resources/Locale/ru-RU/Imperial/XxRaay/preset-syndiebattle.ftl

Co-authored-by: Ser11y <160628372+Ser1-1y@users.noreply.github.com>

XxRaay committed:
refactor: remove PlayerId from SyndieBattleScoreComponent and clean up related code

itzfluffycat commented:
ура наконец-то возвращение легендарного режима

XxRaay committed:
feat: enhance SyndieBattle scoring system with player survival tracking and detailed round end summaries

XxRaay committed:
Merge branch 'develop' of https://github.com/XxRaay/SS14-public-edit into develop

XxRaay committed:
fix: update SyndieBattle redemption machine spawning logic and localization

- Refactored the spawning logic to utilize station data more effectively.
- Adjusted localization for the SyndieBattle Redemption Machine description.

XxRaay committed:
feat: add pacifism mechanics and localization for SyndieBattle

XxRaay committed:
feat: add goal message for SyndieBattle mode and update localization

XxRaay committed:
feat: enhance Supermatter integrity processing and add catastrophe warnings in localization

XxRaay committed:
Revert "feat: enhance Supermatter integrity processing and add catastrophe warnings in localization"

This reverts commit 81f9d7e3fa9d45055f22dffd573fba37f46cd68e.

XxRaay committed:
feat: update payout calculation in SyndieBattle redemption system and add pacifism duration handling

XxRaay committed:
feat: update redemption machine count and implement automatic rule end after 30 minutes in SyndieBattle

XxRaay committed:
refactor: remove redundant chat message for skipped spawn in SyndieBattle redemption machine

XxRaay committed:
refactor: remove redundant cable and pipe checks in SpawnRedemptionMachines method

Ser1-1y commented on Resources/Locale/ru-RU/Imperial/XxRaay/preset-syndiebattle.ftl:
```diff
@@ -0,0 +1,23 @@
+syndiebattle-redemption-no-uplink = Машина вернула КПК!
+syndiebattle-redemption-success = Машина выдала { $amount } предметов!
+syndiebattle-redemption-error = Ошибка при выдаче награды
+syndiebattle-pacify-ended = Период пацифизма завершён.
+syndiebattle-round-end-title = [color=yellow][b]Итоги боя предателей[/b][/color]
+syndiebattle-top-tc-header = [color=orange]Топ-3 по ТК (P.S.: Учитывается в аплинке):[/color]
+syndiebattle-top-tc-entry = [b]{ $rank }.[/b] { $name } — { $tc } ТК
+syndiebattle-top-kills-header = [color=red]Топ-3 по убийствам:[/color]
+syndiebattle-top-kills-entry = [b]{ $rank }.[/b] { $name } — { $kills } убийств
+syndiebattle-top-survive-header = [color=green]Топ-3 по выживанию:[/color]
+syndiebattle-top-survive-entry = [b]{ $rank }.[/b] { $name } — { $time } — { $status }
+syndiebattle-survive-status-alive = Выжил
+syndiebattle-survive-status-dead = Погиб
+syndiebattle-mode-goal = [color=red][b]Цель режима: убивайте других агентов Синдиката! В конце раунда будет топ 3 людей по ТК(ВАЖНО: ТК считается ТОЛЬКО которые в аплинке КПК), убийствам и по времени выживания.[/b][/color]
```
```suggestion
syndiebattle-pacify-ended = Пацифизм кончился!
syndiebattle-round-end-title = [color=yellow][b]Итоги боя предателей[/b][/color]
syndiebattle-top-tc-header = [color=orange]Топ-3 по ТК:[/color]
syndiebattle-top-tc-entry = [b]{ $rank }.[/b] { $name } — { $tc } ТК
syndiebattle-top-kills-header = [color=red]Топ-3 по убийствам:[/color]
syndiebattle-top-kills-entry = [b]{ $rank }.[/b] { $name } — { $kills } убийств
syndiebattle-top-survive-header = [color=green]Топ-3 по выживанию:[/color]
syndiebattle-top-survive-entry = [b]{ $rank }.[/b] { $name } — { $time } — { $status }
syndiebattle-survive-status-alive = Выжил
syndiebattle-survive-status-dead = Погиб
syndiebattle-mode-goal = [color=red][b]Убивайте других агентов синдиката и собирайте их ТК в свой аплинк![/b][/color]
```

Ser1-1y committed:
Small system refactor

Ser1-1y committed:
Cleanup, redemption machine spawn refactor

(cherry picked from commit 057baff7a21a8f9172fd70bff170dc624fb3c072)

Ser1-1y committed:
small cleanup

(cherry picked from commit 69a1ed99bed9c4bc12b98bd6b20e42bc63f78d02)

XxRaay committed:
Delete Resources/Prototypes/Imperial/Entities/Structures/Power/Generation/SuperMatter/supermatter_explosion.yml

XxRaay commented on Content.Server/Imperial/XxRaay/SyndieBattle/SyndieBattleRuleSystem.cs:
```diff
@@ -0,0 +1,411 @@
+            false,
+            false);
+
+        ConvertAllCurrentPlayers(component);
+    }
```
Зачем два раза накладывать? При начале раунда и при спавне игрока. Это не целесообразно

XxRaay committed:
Delete Content.Server/Imperial/XxRaay/SyndieBattle/SyndieBattlePacifiedMarkerComponent.cs
